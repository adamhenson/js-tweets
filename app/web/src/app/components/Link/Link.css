.link {
  color: var(--color-primary-dark);
  text-decoration: none;
}

.activeLink {
  text-decoration: underline;
}