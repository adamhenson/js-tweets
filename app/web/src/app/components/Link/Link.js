import React from 'react';
import PropTypes from 'prop-types';
import { Link as RouterLink, NavLink as RouterNavLink } from 'react-router-dom';
import classnames from 'classnames';
import style from './Link.css';

const Link = ({
  children,
  className,
  isNavLink,
  onClick,
  to,
}) => ((!isNavLink)
  ? (
    <RouterLink
      className={classnames(className, style.link)}
      onClick={onClick}
      to={to}
    >
      {children}
    </RouterLink>
    )
  : (
    <RouterNavLink
      activeClassName={style.activeLink}
      className={classnames(className, style.link)}
      exact
      onClick={onClick}
      to={to}
    >
      {children}
    </RouterNavLink>
  ));

Link.propTypes = {
  isNavLink: PropTypes.bool,
  children: PropTypes.node,
  className: PropTypes.string,
  onClick: PropTypes.func,
  to: PropTypes.string.isRequired,
};

Link.defaultProps = {
  isNavLink: false,
  children: '',
  className: null,
  onClick: null,
};

export default Link;
