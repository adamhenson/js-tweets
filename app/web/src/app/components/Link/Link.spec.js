import React from 'react';
import Link from './Link';

test('Render a router link', () => {
  const wrapper = shallow(
    <Link to="http://www.google.com">foo bar</Link>
  );
  expect(wrapper.prop('to')).toEqual('http://www.google.com');
});

test('Render a router nav link', () => {
  const wrapper = shallow(
    <Link to="http://www.google.com" isNavLink>foo bar</Link>
  );
  expect(wrapper).toMatchSnapshot();
});
