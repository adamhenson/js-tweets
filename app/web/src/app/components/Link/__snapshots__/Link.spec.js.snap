// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render a router nav link 1`] = `
<NavLink
  activeClassName="activeLink"
  className="link"
  exact={true}
  onClick={null}
  to="http://www.google.com"
>
  foo bar
</NavLink>
`;
